domains/transfer-agreements/TransferAgreements.API/API.UnitTests/Controllers/TransferAgreementsControllerTests.cs

```diff
@@ -55,4 +61,37 @@ public async Task Create_ShouldCallServiceOnce()
             agreement.ReceiverTin == request.ReceiverTin
         )), Times.Once);
     }
+
+    [Fact]
+    public async Task GetBySubjectId_ShouldCallServiceOnce()
+    {
+        var userId = Guid.Parse(controller.ControllerContext.HttpContext.User.FindFirstValue("sub") ?? string.Empty);
+
+        await controller.GetBySubjectId();
+
+        mockTransferAgreementRepository.Verify(service => service.GetTransferAgreementsBySubjectId(userId), Times.Once);
+    }
+
+    [Fact]
+    public async Task GetBySubjectId_ShouldReturnCorrectNumberOfAgreements()
+    {
+        var userId = Guid.Parse(controller.ControllerContext.HttpContext.User.FindFirstValue("sub") ?? string.Empty);
```

Comment:
You are mocking the subject in the constructor. I would suggest that you just put it in a variable and use that instead of fecthing from the httpcontext